### DISCUSSION TITLE
Current Yield

### DISCUSSION THREAD

**Discussion by mmo82** [wlcrs/huawei_solar#776] [Ideas]
hi,

Is there any option to get the current Yield from the panels? I only see the hourly yield which increase as the production increases during the hours.
If we would have the current Yield , we could then substract the current consumption of the house. This way we have the available yield ( what is fed to network or batteries) and use it as a trigger for other electrical appliances.

Thanks!

**Comment by BradleyFord** [comment]
Yes yes yes, this is what I have been looking to do as well.
I think I have found the registers I just need to add them on my system, but I've been very busy the last 2 weeks traveling for work. 
https://github.com/wlcrs/huawei_solar/issues/769#issuecomment-2233190569

In my case we need to have the zero export enabled, which means when demand => supply, then we don't know how much more supply is available.

**Reply by Roving-Ronin** [reply]
@mmo82 
The 'current yield' of the panels is summed up as the Input Power of an inverter. The Active Power is then the amount of Power that the inverter is outputing from that, which is usually less a few watts due to the inverter needing power to run itself.

See:
  sensor.inverter_input_power
  sensor.inverter_active_power

The Active Power is 'affected' by the Discharge of Power from a Battery, if you have one.  If your after House Consumption (this is based on inverterS, so edit back to a single inverter sensor if you only have 1 inverter):

```
  # Shows the calculated current load, in Watts
  - sensor:
    - name: "House Consumption Power"
      unique_id: house_consumption_power
      unit_of_measurement: "W"
      icon: mdi:solar-power
      device_class: "power"
      state_class: measurement
      state: >-
        {% set inv_active_power = states('sensor.inverters_active_power')|float(0) %}
        {% set pm_active_power = states('sensor.power_meter_active_power')|float(0) %}
        {{ (inv_active_power - pm_active_power)|float(0)|round(0) }}
      availability: >-
        {{ (states('sensor.inverters_active_power')|is_number)
            and (states('sensor.power_meter_active_power')|is_number) }}

```
__________

@BradleyFord 

If you want to just limit your inverter to ZERO EXPORTS all the time, just run this service in HA Developer Tools: "Huawei Solar: Set Active Power Control to 'Zero power grid connection'"

Can also be set from the Fusion Solar web application, under the sDongle settings.
https://forum.huawei.com/enterprise/en/setting-zero-power-grid-tied-in-the-smart-pv-management-system-smartlogger-scenario/thread/667265450205134848-667213868771979264

Most of this stuff is in the Wiki.... https://github.com/wlcrs/huawei_solar/wiki/Connecting-to-daisy-chained-inverters#combining-multiple-inverter-values
